internal/schemautil/service.go

```diff
@@ -636,12 +636,17 @@ func copyServicePropertiesFromAPIResponseToTerraform(
 		}
 	}
 
-	userConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
+	newUserConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
 	if err != nil {
 		return err
 	}
 
-	if err := d.Set(serviceType+"_user_config", NormalizeIpFilter(d.Get(serviceType+"_user_config"), userConfig)); err != nil {
+	// Mutates user config inplace
+	oldUserConfig := d.Get(serviceType + "_user_config")
+	copySensitiveFields(oldUserConfig, newUserConfig)
```

Comment:
I'm not sure whether this is an idiomatic Go approach, e.g. the function `append` does not mutates data, but returns the new version of it instead

Reply:
NormalizeIpFilter mutates data. See comment below.

Reply:
I won't modify NormalizeIpFilter. It has quite confusing code and doesn't provide extra tests for that complexity. Just check out first few 20-30 lines and early exits.